bleak/agent.py

```diff
@@ -0,0 +1,27 @@
+import abc
+
+from .backends.device import BLEDevice
+
+
+class BaseBleakAgentCallbacks(abc.ABC):
+    @abc.abstractmethod
```

Comment:
Making methods abstract here isn't going to be future-proof. 

We should consider what it would take to implement the various capabilities, e.g. if you claim to support capability X then you need to implement method y().

All methods can have a default implementation that raises `NotImplementedError`. And we can make an `__init__(self, capabilities: BleakAgentFlags)` method that verifies that the required methods have been overridden.